messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistSdk.kt

```diff
@@ -134,6 +135,20 @@ object GistSdk : Application.ActivityLifecycleCallbacks {
         currentRoute = ""
     }
 
+    private fun subscribeToLifecycleEvents() {
+        SDKComponent.activityLifecycleCallbacks.subscribe { events ->
+            events.filter { state ->
+                state.event == Lifecycle.Event.ON_RESUME || state.event == Lifecycle.Event.ON_PAUSE
+            }.collect { state ->
+                when (state.event) {
+                    Lifecycle.Event.ON_RESUME -> onActivityResumed(state.activity)
+                    Lifecycle.Event.ON_PAUSE -> onActivityPaused(state.activity)
```

Comment:
These are not on main thread now, but looking at code, I think we don't need them to be on main specifically? 🤔